mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -412,11 +412,10 @@ class LinearRegressionModel private[ml] (
   def hasSummary: Boolean = trainingSummary.isDefined
 
   /**
-   * Evaluates the model on a testset.
+   * Evaluates the model on a test dataset.
    * @param dataset Test dataset to evaluate model on.
    */
-  // TODO: decide on a good name before exposing to public API
-  private[regression] def evaluate(dataset: DataFrame): LinearRegressionSummary = {
+  def evaluate(dataset: DataFrame): LinearRegressionSummary = {
     // Handle possible missing or invalid prediction columns
     val (summaryModel, predictionColName) = findSummaryModelAndPredictionCol()
     new LinearRegressionSummary(summaryModel.transform(dataset), predictionColName,
```

Comment:
I think that `summaryModel` should be passed into the `LinearRegressionSummary` constructor instead of `this`.  I think I see the predictionCol used in `devianceResiduals`, so if it was just added in the `summaryModel`, it might fail.